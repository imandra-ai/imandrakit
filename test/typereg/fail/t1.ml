let dump () =
  Imandrakit_typereg.(
    to_iter top |> Iter.to_rev_list
    |> List.sort Ty_def.compare_by_name
    |> List.iter (fun d -> Format.printf "%a@." Ty_def.pp d))

module Yolo = struct
  type t = float option list array [@@deriving typereg]
end

module Yolo2 = struct
  type t = float option list array [@@deriving typereg]
end

let () =
  print_endline "t1 fail:";
  dump ()
