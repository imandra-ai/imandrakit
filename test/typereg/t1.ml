open Util_

type foo = int [@@deriving typereg]
type bar = foo * int list option [@@deriving typereg]

module Yolo = struct
  type t = float option list array
  [@@deriving typereg] [@@typereg.name "Yolo.t"]
end

let () =
  print_endline "t1:";
  dump ()
