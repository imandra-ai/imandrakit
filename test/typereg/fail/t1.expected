Fatal error: exception Failure("typereg collision in '_none_': type def \"t\" in \"Dune__exe__T1\" is already present.")
